state_machine/docs/README.md

```diff
@@ -0,0 +1,66 @@
+# State Machine Module
+
+This document outlines the purpose of this module, its components and how they all interact with the other modules.
+
+## Contents <!-- omit in toc -->
+- [State Machine Module](#state-machine-module)
+  - [Overview](#overview)
+  - [Code Structure](#code-structure)
+  - [High Level Architecture](#high-level-architecture)
+    - [FSM primer](#fsm-primer)
+  - [Current State Machine Definition](#current-state-machine-definition)
+
+## Overview
+
+The `StateMachine` module implements a FSM (Finite State Machine) that is responsible for managing the node lifecycle since its internal behaviour can be different depending on certain conditions that are used to determine the current state.
+
+TL;DR: When you are in a state, you can only transition to other states that are allowed by the fsm definition.
+Transitions happen reacting to events.
+
+
+## Code Structure
+
+```bash
+├── docs
+│   ├── CHANGELOG.md              # Changelog
+│   ├── README.md                 # You are here
+│   └── state-machine.diagram.md  # State machine diagram (generated by visualizer/main.go)
+├── fsm.go                        # Finite State Machine definition (events, states, transitions)
+├── module.go                     # Implementation of the StateMachine module
+└── visualizer
+    └── main.go                   # State machine diagram generator
+```
+
+## High Level Architecture
+
+⚠️ Work in progress ⚠️ - At the time of writing this document, the architecture is still being defined.
+
+This is pretty much a POC at the moment.
+
+The key things to keep in mind are:
+
+- We are using the library "github.com/looplab/fsm" to implement the FSM, essentially here we are wrapping it and integrating it with the rest of the modules. This document won't cover the implementation details of the FSM library which is fairly well documented.
+- The **StateMachineModule** can be accessed via the `bus` from any other `IntegratableModule`
```

Comment:
1. I do think we should rename `IntegratableModule` to `RegistrableModule`

Reply:
I detailed why we shouldn't in the other similar comment.